### PR TITLE
New API overview: added AsDynamic call

### PR DESCRIPTION
**Pull request by pkulikov**
The example at the end of the [overview of new ML.NET API](https://github.com/dotnet/machinelearning/blob/master/docs/code/MlNetHighLevelConcepts.md#prediction-function) doesn't use `AsDynamic`:
```csharp
var inputData = env.CreateDataView(new InputExample[] { example });
var outputData = model.Transform(inputData);
var output = outputData.AsEnumerable<OutputPrediction>(env, reuseRowObject: false).Single();
```

So, it looks that the `AsDynamic` call in the cookbook example is not necessary.

### CODE REVIEW

**Comment by pkulikov** [comment]
@Zruty0 OK, I'll make the update in the overview.
Why is this `AsDynamic` call necessary, by the way?

**Comment by pkulikov** [comment]
@Zruty0 I've updated the overview example.

File: docs/code/MlNetCookBook.md [hunk 1]
```diff
@@ -982,4 +982,4 @@ dynamicPipe = dynamicPipe.Append(new KeyToValueEstimator(env, "PredictedLabel"))
 var dynamicModel = dynamicPipe.Fit(data.AsDynamic);
 
 // Now 'dynamicModel', and 'model.AsDynamic' are equivalent.
-```
\ No newline at end of file
+```
```
Comment (Zruty0) [inline docs/code/MlNetCookBook.md]: >`` [](start = 1, length = 2)

nitpicking: please remove the trailing whitespace :)

File: docs/code/MlNetHighLevelConcepts.md [hunk 2]
```diff
@@ -150,4 +150,4 @@ var output = predictionFunc.Predict(example);
 
 The same `predictionFunc` can (and should!) be used multiple times, thus amortizing the initial cost of `MakePredictionFunction` call. 
 
-The prediction function is *not re-entrant / thread-safe*: if you want to conduct predictions simultaneously with multiple threads, you need to have a prediction function per thread.
\ No newline at end of file
+The prediction function is *not re-entrant / thread-safe*: if you want to conduct predictions simultaneously with multiple threads, you need to have a prediction function per thread.
```
Comment (Zruty0) [inline docs/code/MlNetHighLevelConcepts.md]: >d. [](start = 180, length = 2)

Trailing whitespace again :)

File: docs/code/MlNetCookBook.md (hunk 1 above)
Comment (pkulikov) [reply docs/code/MlNetCookBook.md]: @Zruty0 fixed both :). Seems that GitHub web interface adds a blank line at the end on a commit.

Comment (Zruty0) [reply docs/code/MlNetCookBook.md]: Oh I see. Well, I suppose we'll end up with these anyway, if that's what automation is doing. In any case, thanks!

---
In reply to: [222460113](https://github.com/dotnet/machinelearning/pull/1139#discussion_r222460113) [](ancestors = 222460113)

**Pull request merged**